### ISSUE TITLE
[Bug]: Cart Sidebar -  cannot remove product or add more items  

### ISSUE THREAD

**Issue by AdamPawlinski** [vuestorefront-community/vendure#167]
### Expected Behavior

Products on Cart Sidebar element should behave as expected: 
- can be removed from Cart
- can add more items
- 'save for later' button can be removed from this view I suppose (to be discussed) 

### Actual Behavior

![Uploading vendure_cart_preview.gif…]()

### What version of vendure integration are you using?

1.0.0.

### What version of Node.js are you using?

16.13.2

### What browser (and version) are you using?

Chrome Version 98.0.4758.80 (Official Build) (64-bit)

### What operating system (and version) are you using?

Ubuntu

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Baroshem** [comment]
Hey, could you please check what is going on in the network/console tab of the browser? Also could you take a look at Vendure server to investigate whether the changes to the cart are actually happening on Vendure?

**Comment by jacobfrantz1** [comment]
I ran into this issue as well. This integration transforms the bearer token that vendure sets for the guest user when they create an order into a cookie. The cookie has the "Secure" option so chrome won't set the cookie if you are not using https. Having the cookie set is required for any edits to the active order.

**Comment by jacobfrantz1** [comment]
@Baroshem Vendure uses a session token for guest orders. My browser is what rejected the session cookie because it was marked as secure. This is no longer an issue for me as I am now serving my storefront with https in my test environment.
https://github.com/vuestorefront-community/vendure/blob/develop/packages/api-client/src/helpers/cookie.ts
This code marks the cookie as secure.

**Comment by Baroshem** [comment]
I see however, I am also using Chrome for development and serving the dev server locally on http, not https and it is working for some reason 🤔

**Comment by AdamPawlinski** [comment]
@jacobfrantz1 Thanks for solving it, when I changed the cookie to `secure:false` it works. But it is strange that it works properly for @Baroshem without any change. Maybe there is some other difference in your configuration?

**Comment by Baroshem** [comment]
Maybe it has to do with the operating system. But I have seen @michaelbromley creating a demo store while working on Windows. Really really strange. Maybe as a workaround, we could add a condition here so that value of secure will be set to false in dev environment and true in prod?

Something like:

```ts
...
secure: process.env.production ? true : false
```

**Comment by michaelbromley** [comment]
That's right, I use Windows 10 for all my development. Perhaps the issue has to do with the domain. Secure cookeis will only be set over HTTPS *except* when the domain is `localhost` - _in some browsers_. See here for more details: https://web.dev/when-to-use-local-https/#use-http:localhost-by-default

**Comment by macrozone** [comment]
in my case if i increase the amount on an item in the cart, the item just vanishes.

The api returns: "GraphQL error: Die Bestellung beeinhaltet keine OrderLine mit der ID 40"

it means in english something like "The order does not contain OrderLine with ID 40"

**Comment by Baroshem** [comment]
@macrozone 

Could you provide some reproduction repository?

**Comment by macrozone** [comment]
> @macrozone
> 
> Could you provide some reproduction repository?

nevermind, was a cookie error.

the cart is still very buggy though. E.g. after a checkout, the cart is still there, but you cant add any items, it throws an error

```
{"data":{"addItemToOrder":{"errorCode":"ORDER_MODIFICATION_ERROR","message":"Der Inhalt der Bestellung kann nur im Status \"AddingItems\" geändert werden","__typename":"OrderModificationError"}}}

```

**Comment by Baroshem** [comment]
@macrozone 

Could you provide more details (ideally a reproduction repo)? I could try to reproduce it on my side. These issues are really strange because cart operations were there for about few months now, so these issues appear out of nowhere.

**Comment by michaelbromley** [comment]
Once an Order is completed (by default this occurs when it transitions to the PaymentAuthorized or PaymentSettled states), it get set to `active: false` and should no longer be accessible via the `activeOrder` query in the Shop API.

So yes, it sounds like maybe an error is occurring at the payment stage. If you are able to isolate the steps which reproduce it, please open an issue on the Vendure repo.

Regarding error handling in VSF, pretty much all mutations in Vendure will return a union result containing either the expected object _or_ an `ErrorResult` type.

The recommended way to handle this in the storefront is to select and check the `__typename` field when the response from the mutation is returned. If the `__typename` is not the expected type, e.g. `'Order'` then we can assume an error ocurred and display some generic toast notification with the errorMessage.
